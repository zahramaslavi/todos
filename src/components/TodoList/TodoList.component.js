import React from 'react';

import './TodoList.component.scss';

import Todo from '../Todo/Todo.component';

const TodoList = (props) => {

    return (
        <div className="todoList">
            {   props.todos.map(
                        todo => (
                            <Todo 
                                key={todo.id} 
                                todo={todo} 
                                onDeleteTodo={() => props.handleDeleteTodo(todo.id)} 
                                onToggleComplete={() => props.handleToggleComplete(todo.id)}
                            />
                    )
                )
            }
        </div>
    );

}

export default TodoList;