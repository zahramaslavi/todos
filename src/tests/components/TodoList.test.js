import React from 'react';
import { shallow } from 'enzyme';
import TodoList from '../../components/TodoList/TodoList.component';

const data = [
    {description: 'test',completed: false, id: 0},
    {description: 'test',completed: false, id: 1},
    {description: 'test',completed: false, id: 2},
];

const onSubmitSpy = jest.fn();

const wrapper = shallow(
    <TodoList
        todos={data}
        handleDeleteTodo={onSubmitSpy} 
        handleToggleComplete={onSubmitSpy} 
    />);

test('should test TodoList component snapshot', () => {
    expect(wrapper).toMatchSnapshot();
});


const firstTodo = wrapper.find('Todo').first();

test('should test Todos component onDeleteTodo of first todo', () => {
    firstTodo.props().onDeleteTodo();
    expect(onSubmitSpy).toHaveBeenCalled();
});

test('should test Todos component toggleComplete of first todo', () => {
    firstTodo.props().onToggleComplete();
    expect(onSubmitSpy).toHaveBeenCalled();
});