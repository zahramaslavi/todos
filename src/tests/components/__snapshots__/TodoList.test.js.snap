// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test TodoList component snapshot 1`] = `
<div
  className="todoList"
>
  <Todo
    key="1"
    onDeleteTodo={[Function]}
    onToggleComplete={[Function]}
    todo={
      Object {
        "completed": false,
        "description": "test",
        "id": 1,
      }
    }
  />
  <Todo
    key="1"
    onDeleteTodo={[Function]}
    onToggleComplete={[Function]}
    todo={
      Object {
        "completed": false,
        "description": "test",
        "id": 1,
      }
    }
  />
  <Todo
    key="1"
    onDeleteTodo={[Function]}
    onToggleComplete={[Function]}
    todo={
      Object {
        "completed": false,
        "description": "test",
        "id": 1,
      }
    }
  />
</div>
`;
